examples/long-running.rb

```diff
@@ -0,0 +1,63 @@
+$LOAD_PATH.unshift File.expand_path("../../lib", __FILE__)
+require "lowdown"
+require "logger"
+
+cert_file, environment, device_token = ARGV.first(3)
+unless cert_file && environment && device_token
+  puts "Usage: #{$0} path/to/cert.pem [production|development] device-token"
+  exit 1
+end
+
+#$CELLULOID_DEBUG = true
+#Celluloid.logger.level = Logger::DEBUG
+#Celluloid.logger.level = Logger::INFO
+Celluloid.logger.level = Logger::ERROR
+
+logger = Logger.new(STDOUT)
+
+client = Lowdown::Client.production(environment == "production",
+                                    certificate: File.read(cert_file),
+                                    pool_size: 3,
+                                    keep_alive: true) # This option is the key to long running connections
+client.connect
+
+loop do
+  begin
+    logger.info "Perform burst"
+    # Perform a burst of notifications from multiple concurrent threads to demonstrate thread safety.
+    #
+    Array.new(3) do
+      Thread.new do
```

Comment:
To be clear, the actual connection objects do use Celluloid's facilities for concurrency.